tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/SAPXmlHelper.cs

```diff
@@ -1,1113 +1,114 @@
-                    attrNotMatched.Clear();
-                    if (!item.ACTIONNUMBER.Equals(actionCounter.ToString()))
-                        attrNotMatched.Add(nameof(item.ACTIONNUMBER));
-                    //xmlAttributes[1] is skipped as already checked
-                    if (!item.PRODUCT.Equals("AVCS UNIT"))
-                        attrNotMatched.Add(nameof(item.PRODUCT));
-                    if (!item.PRODTYPE.Equals((GetFirstProductsInfoHavingUoS(productName)).ProductType))
-                        attrNotMatched.Add(nameof(item.PRODTYPE));
-                    if (!item.AGENCY.Equals((GetFirstProductsInfoHavingUoS(productName)).Agency))
-                        attrNotMatched.Add(nameof(item.AGENCY));
-                    if (!item.PROVIDER.Equals((GetFirstProductsInfoHavingUoS(productName)).ProviderCode))
-                        attrNotMatched.Add(nameof(item.PROVIDER));
-                    if (!item.ENCSIZE.Equals(unitOfSale.UnitSize))
-                        attrNotMatched.Add(nameof(item.ENCSIZE));
-                    if (!item.TITLE.Equals(unitOfSale.Title))
-                        attrNotMatched.Add(nameof(item.TITLE));
-                    if (!item.UNITTYPE.Equals(unitOfSale.UnitType))
-                        attrNotMatched.Add(nameof(item.UNITTYPE));
-                    VerifyAdditionalXmlTags(item, correctionTag);
-                    //Checking blanks
-                    List<string> blankFieldNames = new() { "CANCELLED", "REPLACEDBY", "EDITIONNO", "UPDATENO", "ACTIVEKEY", "NEXTKEY" };
-                    VerifyBlankFields(item, blankFieldNames);
+                var generatedAction = generatedAttributes[i];
+                var expectedAction = expectedItems[i];
+                string action = generatedAction.Element("ACTION")?.Value;
 
-                    if (attrNotMatched.Count == 0)
-                    {
-                        Console.WriteLine("CREATE AVCS UNIT OF SALE Action's Data is correct");
-                        return true;
-                    }
-
-                    Console.WriteLine("CREATE AVCS UNIT OF SALE Action's Data is incorrect");
-                    Console.WriteLine("Not matching attributes are:");
-                    foreach (string attribute in attrNotMatched)
-                    { Console.WriteLine(attribute); }
-                    return false;
-                }
-            }
-            Console.WriteLine("JSON doesn't have corresponding Unit of Sale.");
-            return false;
-        }
-
-        private static bool VerifyCreateEncCell(string childCell, ZMAT_ACTIONITEMS item, string correctionTag, string permitState)
-        {
-            Console.WriteLine("Action#:" + actionCounter + ".Childcell:" + childCell);
-            foreach (Product product in JsonPayload.Data.Products)
-            {
-                if ((childCell == product.ProductName) && (product.Status.IsNewCell))
+                foreach (var element1 in generatedAction.Elements())
```

Comment:
Can we have meaningful variable name please

Reply:
done